crates/sui-edge-proxy/src/handlers.rs

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::config::PeerConfig;
+use crate::metrics::AppMetrics;
+use axum::{
+    body::Body,
+    extract::{Request, State},
+    http::request::Parts,
+    http::StatusCode,
+    response::Response,
+};
+use bytes::Bytes;
+use std::time::Instant;
+use tracing::{info, warn};
+
+#[derive(Clone)]
+pub struct AppState {
+    client: reqwest::Client,
+    read_peer: PeerConfig,
+    execution_peer: PeerConfig,
+    metrics: AppMetrics,
+}
+
+impl AppState {
+    pub fn new(
+        client: reqwest::Client,
+        read_peer: PeerConfig,
+        execution_peer: PeerConfig,
+        metrics: AppMetrics,
+    ) -> Self {
+        Self {
+            client,
+            read_peer,
+            execution_peer,
+            metrics,
+        }
+    }
+}
+
+pub async fn health_check_handler() -> Response {
+    // TODO: Add more comprehensive health check which checks that the peers are reachable and healthy
+    Response::new("up".into())
+}
+
+pub async fn proxy_handler(
+    State(state): State<AppState>,
+    request: Request<Body>,
+) -> Result<Response, (StatusCode, String)> {
+    let (parts, body) = request.into_parts();
+    if parts.headers.get("content-type")
+        != Some(&reqwest::header::HeaderValue::from_static(
+            "application/json",
+        ))
+    {
+        info!("Content type is not application/json");
+        return Ok(Response::builder()
+            .status(StatusCode::BAD_REQUEST)
+            .body(Body::from("Content type must be application/json"))
+            .unwrap());
+    }
+    let body_bytes = match axum::body::to_bytes(body, usize::MAX).await {
+        Ok(bytes) => bytes,
+        Err(e) => {
+            warn!("Failed to read request body: {}", e);
+            return Ok(Response::builder()
+                .status(StatusCode::INTERNAL_SERVER_ERROR)
+                .body(Body::from("Failed to read request body"))
+                .unwrap());
+        }
+    };
+
+    match parts
+        .headers
+        .get("Client-Request-Method")
```

Comment:
this isn't a standard http header afact, so we should name it `X-Sui-Client-Request-Method` or something, right?

Reply:
is `X-` commonly used for custom headers? I copied the format of other custom headers in the sdk: https://github.com/MystenLabs/sui/blob/main/sdk/typescript/src/client/http-transport.ts#L95-L98